File: src/main/javabak/datagen/ModLootTableProvider.java
package redcrafter07.processed.datagen;

import net.minecraft.core.HolderLookup;
import net.minecraft.data.PackOutput;
import net.minecraft.data.loot.LootTableProvider;
import net.minecraft.world.level.storage.loot.parameters.LootContextParamSets;
import redcrafter07.processed.datagen.loot.ModBlockLootTables;

import java.util.List;
import java.util.Set;
import java.util.concurrent.CompletableFuture;

public class ModLootTableProvider {
    public static LootTableProvider create(PackOutput output, CompletableFuture<HolderLookup.Provider> registries) {
        return new LootTableProvider(
                output, Set.of(),
                List.of(
                        new LootTableProvider.SubProviderEntry(ModBlockLootTables::new, LootContextParamSets.BLOCK)
                ), registries
        );
    }
}

File: src/main/kotlin/redcrafter07/processed/datagen/ModLootTableProvider.kt
package redcrafter07.processed.datagen

import net.minecraft.core.HolderLookup
import net.minecraft.data.PackOutput
import net.minecraft.data.loot.LootTableProvider
import net.minecraft.data.loot.LootTableProvider.SubProviderEntry
import net.minecraft.world.level.storage.loot.parameters.LootContextParamSets
import java.util.concurrent.CompletableFuture

object ModLootTableProvider {
    fun create(output: PackOutput, registries: CompletableFuture<HolderLookup.Provider?>): LootTableProvider {
        return LootTableProvider(
            output,
            mutableSetOf(),
            listOf(SubProviderEntry(::ModBlockLootTables, LootContextParamSets.BLOCK)),
            registries
        )
    }
}